src/app/App.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+import './App.css';
+import Card from '../components/Card/Card';
+import Search from '../components/Search/Search';
+import Filter from '../components/Filter/Filter';
+import getCountries from './countries';
+import type { Country }  from '../models/Country';
+
+function App() {
+  const [countries, setCountries] = useState<Country[]>([]);
+  const [selectedRegion, setSelectedRegion] = useState<string>("");
+  const [searchText, setSearchText] = useState<string>("");
+
+  useEffect(() => {
+      (async () => {
+        const countriesResponse = await getCountries()
+        setCountries(countriesResponse)
+      })();
+  }, []);
+
```

Comment:
You should use a load variable and indicate if the response is loading or it had an error.
`const [isLoading, setIsLoading] = useState<boolean>(true);`